src/components/PeopleFilters.tsx

```diff
@@ -1,21 +1,62 @@
+import classNames from 'classnames';
+import { ChangeEvent } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { getSearchWith } from '../utils/searchHelper';
+import { SearchLink } from './SearchLink';
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const sex = searchParams.get('sex') || '';
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const centuriesValues = ['16', '17', '18', '19', '20'];
+
+  const onQueryChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const preparedSearch = getSearchWith(
+      searchParams,
+      { query: event.target.value.trimStart() || null },
+    );
+
+    setSearchParams(preparedSearch);
+  };
+
   return (
     <nav className="panel">
       <p className="panel-heading">Filters</p>
 
       <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">All</a>
-        <a className="" href="#/people?sex=m">Male</a>
-        <a className="" href="#/people?sex=f">Female</a>
+        <SearchLink
+          className={classNames({ 'is-active': !sex })}
+          params={{ sex: null }}
+        >
+          All
+        </SearchLink>
+
+        <SearchLink
+          className={classNames({ 'is-active': sex === 'm' })}
+          params={{ sex: 'm' }}
```

Comment:
Would be great to create an enum for the sex to avoid errors connected with typos